### PR TITLE
Api fetching

### LINKED ISSUE #53: Login
# User story

As a **customer**,
I want to **login**
so I can access **the content.**

# Notes / Links

# Acceptance criteria

### PR DESCRIPTION
**Pull request by manuelpapa**
In this PR I established the API-connection on the client-side.
You are now able to login and after validation you can access the offers' screen.
It is also possible to create an offer and to reserve another.
In the user section those offers, served and/or created by the user, are shown.
Fix #53 

In the next steps I will include a validation to the create offer process.

### CODE REVIEW

File: client/src/api/results.js [hunk 1]
```diff
@@ -1,148 +1,124 @@
...
+export async function fetchAvailableOffers() {
+  try {
+    const response = await fetch("/api/offers/available", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    const results = await response.json();
+    console.log(results[0]);
```
Comment (lmachens) [inline client/src/api/results.js]:
```suggestion
```

File: client/src/api/results.js [hunk 2]
```diff
@@ -1,148 +1,124 @@
...
+    return results;
+  } catch (error) {
+    alert(error.message);
+    return null;
```
Comment (lmachens) [inline client/src/api/results.js:12]: If you catch the error here, it's difficult to react in your UI to specific errors (show error message).
Instead, I would try-catch in your React components / `useEffect`.
Same issue in other fetches.

File: client/src/api/results.js [hunk 3]
```diff
@@ -1,148 +1,124 @@
...
-      category: "bread",
-      city: "50968 Köln",
-      date: "15.08.2020",
-      time: "14:00 - 15:00",
-    },
-  ];
-  return mockedResults;
+export async function fetchOffers() {
+  try {
+    const response = await fetch("/api/offers/user/offers", {
```
Comment (lmachens) [inline client/src/api/results.js]: This route is a bit uncommon.
If you want to receive offers of the current user, better change it to `/api/user/offers`.
You might face issues with other routes like `/api/offers/:offerId`
Now `user` could be an `offerId` too.

File: client/src/api/results.js [hunk 4]
```diff
@@ -1,148 +1,124 @@
...
+    const request = {
+      title: data.title,
+      category: data.category,
+      location: {
+        name: data.name,
+        street: data.street,
+        city: data.city,
+        zip: data.zip,
+      },
+      start_time: start_time,
```
Comment (lmachens) [inline client/src/api/results.js:95]: you use camelCase in your other code, why start with underscores?

File: client/src/pages/Create.js [hunk 5]
```diff
@@ -29,35 +32,56 @@ const Location = styled.div`
...
+      formattedDate,
+      start_time,
+      end_time,
+      tags
+    );
+    if (response) {
+      history.push("/created");
     }
-  `;
+  };
```
Comment (lmachens) [inline client/src/pages/Create.js]: Some logs and comments you should remove :)

File: client/src/pages/Create.js [hunk 6]
```diff
@@ -29,35 +32,56 @@ const Location = styled.div`
...
+  left: 0;
+  bottom: 0;
+  width: 100vw;
+  background: #fff;
+  & button {
+    margin: 0.5em;
+  }
+`;
 export function Create() {
+  const [value, setValue] = useState("misc");
```
Comment (lmachens) [inline client/src/pages/Create.js]: What is `value`? It's unspecific. It looks like it is used as `category`?

File: client/src/pages/Create.js [hunk 7]
```diff
@@ -29,35 +32,56 @@ const Location = styled.div`
...
+  const [date, setDate] = useState(null);
+  // const [time, setTime] = useState(new Date());
+  const [time, setTime] = useState();
   const { register, handleSubmit, errors } = useForm();
```
Comment (lmachens) [inline client/src/pages/Create.js]: You have a mix of `useForm` states (see usage of `register`) and custom states like `date` or `time`.
It makes it difficult to maintain if you mix it :).

File: client/src/pages/Create.js [hunk 8]
```diff
@@ -138,27 +173,42 @@ export function Create() {
...
+              ref={register({
+                required: true,
+              })}
             />
           </Location>
           <Footer>
             <FooterGradient />
             <Button type="submit">Anbieten</Button>
-            <CancelButton type="reset">Abbrechen</CancelButton>
+            <CancelButton onClick={() => history.goBack()}>
```
Comment (lmachens) [inline client/src/pages/Create.js]: You can optimize this a bit

```suggestion
            <CancelButton onClick={history.goBack}>
```

File: routes/offers.js [hunk 9]
```diff
@@ -1,75 +1,102 @@
...
-const { getOffers, getOffer, createOffer } = require("../lib/offers");
+// Get all offers
+router.get("/", verify, async (req, res) => {
+  try {
+    const offers = await Offer.find();
+    if (!offers) return res.status(400).send("No offers available");
+    res.send(offers);
+  } catch (error) {
+    res.status(500).send("Internal server error");
+  }
```
Comment (lmachens) [inline routes/offers.js:13]: If something unexpected happens, you will not see the error message right now. Happens multiple times.

```suggestion
  } catch (error) {
    console.error(error);
    res.status(500).send("Internal server error");
  }
```

**Review by lmachens — CHANGES_REQUESTED** [review]
Good changes 👍 
You might profit from refactoring your API paths and make them more RESTful.

File: client/src/api/results.js (hunk 3 above)
Comment (manuelpapa) [reply client/src/api/results.js]: Thank you, Leon. Changed it now.

File: client/src/api/results.js (hunk 4 above)
Comment (manuelpapa) [reply client/src/api/results.js:95]: Will change it in further steps. Thank you!

File: routes/offers.js (hunk 9 above)
Comment (manuelpapa) [reply routes/offers.js:13]: Will fix this soon. Thank you so much!

File: client/src/pages/Create.js (hunk 6 above)
Comment (manuelpapa) [reply client/src/pages/Create.js]: Changed it.

**Pull request merged**